private/buf/cmd/buf/command/beta/price/price.go

```diff
@@ -0,0 +1,185 @@
+	return &appcmd.Command{
+		Use:   name + " <source>",
+		Short: "Get the price for BSR paid plans for a given source or module",
+		Long:  bufcli.GetSourceOrModuleLong(`the source or module to get a price for`),
+		Args:  cobra.MaximumNArgs(1),
+		Run: builder.NewRunFunc(
+			func(ctx context.Context, container appflag.Container) error {
+				return run(ctx, container, flags)
+			},
+			bufcli.NewErrorInterceptor(),
+		),
+		BindFlags: flags.Bind,
+	}
+}
+
+type flags struct {
+	DisableSymlinks bool
+
+	// special
+	InputHashtag string
+}
+
+func newFlags() *flags {
+	return &flags{}
+}
+
+func (f *flags) Bind(flagSet *pflag.FlagSet) {
+	bufcli.BindDisableSymlinks(flagSet, &f.DisableSymlinks, disableSymlinksFlagName)
+	bufcli.BindInputHashtag(flagSet, &f.InputHashtag)
+}
+
+func run(
+	ctx context.Context,
+	container appflag.Container,
+	flags *flags,
+) error {
+	input, err := bufcli.GetInputValue(container, flags.InputHashtag, ".")
+	if err != nil {
+		return err
+	}
+	sourceOrModuleRef, err := buffetch.NewRefParser(container.Logger(), buffetch.RefParserWithProtoFileRefAllowed()).GetSourceOrModuleRef(ctx, input)
+	if err != nil {
+		return err
+	}
+	storageosProvider := bufcli.NewStorageosProvider(flags.DisableSymlinks)
+	runner := command.NewRunner()
+	clientConfig, err := bufcli.NewConnectClientConfig(container)
+	if err != nil {
+		return err
+	}
+	moduleReader, err := bufcli.NewModuleReaderAndCreateCacheDirs(container, clientConfig)
+	if err != nil {
+		return err
+	}
+	moduleConfigReader, err := bufcli.NewWireModuleConfigReaderForModuleReader(
+		container,
+		storageosProvider,
+		runner,
+		clientConfig,
+		moduleReader,
+	)
+	if err != nil {
+		return err
+	}
+	moduleConfigs, err := moduleConfigReader.GetModuleConfigs(
+		ctx,
+		container,
+		sourceOrModuleRef,
+		"",
+		nil,
+		nil,
+		false,
+	)
+	if err != nil {
+		return err
+	}
+	statsSlice := make([]*protostat.Stats, len(moduleConfigs))
+	for i, moduleConfig := range moduleConfigs {
+		stats, err := protostat.GetStats(ctx, bufmodulestat.NewFileWalker(moduleConfig.Module()))
+		if err != nil {
+			return err
+		}
+		statsSlice[i] = stats
+	}
+	tmpl, err := template.New("tmpl").Parse(tmplCopy)
+	if err != nil {
+		return err
+	}
+	return tmpl.Execute(
+		container.Stdout(),
+		newTmplData(protostat.MergeStats(statsSlice...)),
+	)
+}
+
+type tmplData struct {
+	*protostat.Stats
+
+	NumTypes             int
+	ChargeableTypes      int
+	ChargeableIsRounded  bool
+	TeamsDollarsPerMonth string
+	ProDollarsPerMonth   string
+}
+
+func newTmplData(stats *protostat.Stats) *tmplData {
+	tmplData := &tmplData{
+		Stats:    stats,
+		NumTypes: stats.NumMessages + stats.NumEnums + stats.NumMethods,
+	}
+	buckets := tmplData.NumTypes / 100
+	if tmplData.NumTypes%100 != 0 {
+		tmplData.ChargeableIsRounded = true
+		buckets++
+	}
+	tmplData.ChargeableTypes = buckets * 100
+	tmplData.TeamsDollarsPerMonth = fmt.Sprintf("%.2f", float64(tmplData.ChargeableTypes)*teamsPerTypeDollarsPerMonth)
```

Comment:
Not to be over engineering but if we do any more advanced stuff with money calculations we should probably use a decimal/money package as to not have rounding errors down the line